lib/ansible/plugins/inventory/hcloud.py

```diff
@@ -168,16 +168,33 @@ def _set_server_attributes(self, server):
             self.inventory.set_variable(server.name, "ansible_host", to_native(server.public_net.ipv4.dns_ptr))
 
         # Server Type
-        self.inventory.set_variable(server.name, "server_type", to_native(server.image.name))
+        if server.image is not None and server.image.name is not None:
+            self.inventory.set_variable(server.name, "server_type", to_native(server.image.name))
+        else:
+            self.inventory.set_variable(server.name, "server_type", to_native("No Image name found."))
 
         # Datacenter
         self.inventory.set_variable(server.name, "datacenter", to_native(server.datacenter.name))
         self.inventory.set_variable(server.name, "location", to_native(server.datacenter.location.name))
 
         # Image
-        self.inventory.set_variable(server.name, "image_id", to_native(server.image.id))
-        self.inventory.set_variable(server.name, "image_name", to_native(server.image.name))
-        self.inventory.set_variable(server.name, "image_os_flavor", to_native(server.image.os_flavor))
+        if server.image is not None:
+            if server.image.id is not None:
+                self.inventory.set_variable(server.name, "image_id", to_native(server.image.id))
+            else:
+                self.inventory.set_variable(server.name, "image_id", to_native("No Image ID found"))
+            if server.image.name is not None:
+                self.inventory.set_variable(server.name, "image_name", to_native(server.image.name))
```

Comment:
The name of an image can be None, if the name is None you should go on the non unique description.

Images with names are only our system images. Without names are backups or snapshots.